app/src/main/res/layout/app_bar_main.xml

```diff
@@ -20,7 +20,22 @@
 
     </android.support.design.widget.AppBarLayout>
 
-    <include layout="@layout/content_main" />
+    <android.support.v4.view.ViewPager
+        android:id="@+id/viewpager"
+        app:layout_behavior="@string/appbar_scrolling_view_behavior"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <android.support.design.widget.TabLayout
```

Comment:
The TabLayout should be a child of the AppBarLayout, not the ViewPager. And add an id to the AppBarLayout so it can be used as an anchor to the ViewPager, to make its contents appear starting from below the AppBarLayout.

```xml
<android.support.design.widget.AppBarLayout
        android:id="@+id/appBarLayout"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:theme="@style/AppTheme.AppBarOverlay">

        <android.support.v7.widget.Toolbar
            android:id="@+id/toolbar"
            android:layout_width="match_parent"
            android:layout_height="?attr/actionBarSize"
            android:background="?attr/colorPrimary"
            app:popupTheme="@style/AppTheme.PopupOverlay" />

        <android.support.design.widget.TabLayout
            android:id="@+id/tabs"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:layout_gravity="top"
            android:background="@color/colorPrimary"
            app:tabBackground="@color/colorPrimaryDark"
            app:tabIndicatorColor="@color/colorAccent" />

    </android.support.design.widget.AppBarLayout>

    <android.support.v4.view.ViewPager
        android:id="@+id/viewpager"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        app:layout_behavior="@string/appbar_scrolling_view_behavior"
        app:layout_anchor="@id/appBarLayout">

    </android.support.v4.view.ViewPager>
```

Reply:
Thanks for the comments. I wanted to fix the problem with the tabs that were sometimes appearing and sometimes not so I followed this link:
https://material.io/develop/android/components/tab-layout/ and it was working.
But this works too so I will add the changes.

Reply:
did these changes, fix that problem where we had to clean and rebuild and hammer the projects lol?